tealish/nodes.py

```diff
@@ -360,73 +397,76 @@ class Exit(LineStatement):
     name: str
     expression: GenericExpression
 
-    def process(self):
+    def process(self) -> None:
         self.expression.process()
 
-    def write_teal(self, writer):
+    def write_teal(self, writer: "TealWriter") -> None:
         writer.write(self, f"// {self.line}")
         writer.write(self, self.expression)
         writer.write(self, "return")
 
-    def _tealish(self, formatter=None):
-        return f"exit({self.expression.tealish(formatter)})\n"
+    def _tealish(self) -> str:
+        return f"exit({self.expression.tealish()})\n"
 
 
 class FunctionCallStatement(LineStatement):
     pattern = r"(?P<expression>[a-zA-Z_0-9]+\(.*\))$"
     expression: GenericExpression
 
-    def process(self):
+    def process(self) -> None:
         self.expression.process()
-        self.name = self.expression.name
+        # TODO: wat?
+        self.name = self.expression.get_current_scope()["name"]
         if self.expression.type:
             raise CompileError(
                 f"Unconsumed return values ({self.expression.type}) from {self.name}",
                 node=self,
             )
 
-    def write_teal(self, writer):
+    def write_teal(self, writer: "TealWriter") -> None:
         writer.write(self, f"// {self.line}")
         writer.write(self, self.expression)
 
-    def _tealish(self, formatter=None):
-        return f"{self.expression.tealish(formatter)}\n"
+    def _tealish(self) -> str:
+        return f"{self.expression.tealish()}\n"
 
 
 class Assert(LineStatement):
     pattern = r'assert\((?P<arg>.*?)(, "(?P<message>.*?)")?\)$'
     arg: GenericExpression
     message: str
 
-    def process(self):
+    def process(self) -> None:
         self.arg.process()
         if self.arg.type not in ("int", "any"):
             raise CompileError(
                 f"Incorrect type for assert. Expected int, got {self.arg.type} at line {self.line_no}.",
                 node=self,
             )
-        if self.message:
+
+        # TODO: added check for compiler not None, should it ever happen that it is None?
+        if self.message and self.compiler is not None:
```

Comment:
we need to do this check for compiler none since the original Node marks it optional